transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollEventLoop.java

```diff
@@ -377,11 +378,15 @@ protected void run() {
                         break;
 
                     case SelectStrategy.SELECT:
-                        if (wakenUp.get() == 1) {
-                            wakenUp.set(0);
-                        }
-                        if (!hasTasks()) {
-                            strategy = epollWait();
+                        wakenUp.lazySet(0);
+                        try {
+                            if (!hasTasks()) {
+                                strategy = epollWait();
+                            }
+                        } finally {
+                            if (wakenUp.get() == 1 || wakenUp.getAndSet(1) == 1) {
```

Comment:
also why the get first ? To reduce the cost of the CAS ?

Reply:
@normanmaurer yes this is a big saving in the case the EL was woken due to task submission (at least an order of magnitude IIRC). So unless timer/IO wakeups are _much_ more frequent overall (possible?) this seems worthwhile. I'd be happy either way though.

This may seem contrary to my stance in #9109 but I wasn't disagreeing with the proposal there, rather suggesting that we shouldn't change an _existing_ usage unless we're sure it's beneficial.

Reply:
please add a comment as well :)